src/Form/Type/ImmutableTabsType.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ *
+ * This file is part of the Sonata for Ekino project.
+ *
+ * (c) 2018 - Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ */
+
+namespace Sonata\SonataHelpersBundle\Form\Type;
+
+use Sonata\CoreBundle\Form\Type\ImmutableArrayType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\Component\Form\FormView;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class ImmutableTabsType.
+ *
+ * @author Benoit de Jacobet <benoit.de-jacobet@ekino.com>
+ */
+class ImmutableTabsType extends ImmutableArrayType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $resolver->setRequired('tabs');
+    }
+
+    /**
+     * @param FormBuilderInterface $builder
+     * @param array                $options
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $tabKeys = $options['keys'];
+
+        // Tabs
+        foreach ($options['tabs'] as $tabCode => $tab) {
+            $builder->add($tabCode, ImmutableArrayType::class, [
```

Comment:
if you have several ImmutableTabsType maybe we should prefix the tabCode with the parent name,  to avoid conflicts, like several 'fr' fields that would ovewritte each others ?

Reply:
Sry, forget it, the tree structure prevents this